admin_guide/custom_controllers.adoc

```diff
@@ -0,0 +1,302 @@
+[[admin-guide-custom-controllers]]
+= Creating Custom Controllers
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+
+
+In the Kubernetes API, a controller is a code loop that watches the state of a cluster 
+through the API server and makes changes in an attempt to move the current state towards the desired state. 
+Examples of controllers that ship with {product-title} today are:
+
+* xref:../dev_guide/daemonsets.adoc#dev-guide-daemonsets[DaemonSets] make sure that a pod is running on each node.
+* The xref:../admin_guide/scheduling/scheduler.adoc#admin-guide-scheduler[scheduler] makes sure that every pod is assigned to a node.
+* The endpoint controller makes sure that for every pod that belongs to a service, the pod IP address is in the endpoints of the service.
+* The physical volume controller makes sure that for every Persistent Volume Claim there is an actual volume created with the cloud provider.
+
+xref:../architecture/core_concepts/deployments.adoc#replication-controllers[replication controllers].
+
+You can create custom controllers using the `oc observe` command. The commmand allows you to run scripts against a 
+cluster that react to changes in the cluster. The command is available as an image named `openshift/observe:latest`.
+
+//from man page
+On startup, `oc observe` lists all of the resources of a particular type and executes the provided script on each one. 
+The command watches the server for changes, and will re-execute the script for each update.
+
+The `oc observe` command works best for problems of the form _for every resource X, make sure Y is true_. Some
+examples of ways observe can be used include: 
+
+  * Ensuring every namespace has a quota or limit range object  
+  * Ensuring every service is registered in DNS by making calls to a DNS API  
+  * Sending an email alert whenever a node reports 'NotReady'  
+  * Watching for the 'FailedScheduling' event and write an IRC message  
+  * Dynamically provisioning persistent volumes when a new PVC is created  
+  * Deleting pods that have reached successful completion after a period of time.  
+
+=== Installing `oc observe`
+
+To obtain and run the `oc observe` command:
+
+. Run the following command to bind mount a kubeconfig as a volume:
++
+----
+docker run -v /var/run/kubernetes/admin.kubeconfig:/kubeconfig -e KUBECONFIG=/kubeconfig openshift/observe:latest services
```

Comment:
sorry, I was giving that path as an example and it is `hack/local-up-cluster.sh` based.

Lets use the originified path that would work on the master:

`docker run -u root -v /etc/origin/master/admin.kubeconfig:/kubeconfig -e KUBECONFIG=/kubeconfig openshift/observe:latest services`

This works with SELinux enforcing on the master; I tried it.

Reply:
@sjenning I run the above command and I get:
```docker run -u root -v /etc/origin/master/admin.kubecoig:/kubeconfig -e KUBECONFIG=/kubeconfig openshift/observe:latest services
# 2018-01-18T00:43:15Z Sync started
# 2018-01-18T00:43:15Z Sync 1068	"" default docker-registry
# 2018-01-18T00:43:15Z Sync 433	"" default kubernetes
# 2018-01-18T00:43:15Z Sync 36694	"" default proxy
# 2018-01-18T00:43:15Z Sync 1162	"" default registry-console
# 2018-01-18T00:43:15Z Sync 998	"" default router```

Is this expected? What does the user do at this point? There is no prompt to do anything further.